spec/services/calculators/childcare_eligibility_calculator_spec.rb

```diff
@@ -4,58 +4,76 @@ module Calculators
-          expect(calculated_result).to eq false
-        end
+      let(:partner_gross_income_subtotals) do
+        instance_double(PersonGrossIncomeSubtotals, employment_income_subtotals: instance_double(EmploymentIncomeSubtotals, in_work?: true))
       end
 
-      context "with child dependants, an employed applicant and no partner" do
-        let(:dependants) { [instance_double(Dependant, becomes_16_on: submission_date + 1.year)] }
+      context "without partner" do
+        subject(:calculated_result) { described_class.call(applicant_incomes: [applicant_gross_income_subtotals].compact, dependants:, submission_date:) }
 
-        it "returns true" do
-          expect(calculated_result).to eq true
+        context "with no dependants, an employed applicant and no partner" do
+          it "returns false" do
+            expect(calculated_result).to eq false
+          end
         end
-      end
 
-      context "with child dependants, a student applicant and no partner" do
-        let(:dependants) { [instance_double(Dependant, becomes_16_on: submission_date + 1.year)] }
-        let(:applicant) { OpenStruct.new(in_work?: false, is_student?: true) }
+        context "with child dependants, an employed applicant and no partner" do
+          let(:dependants) { [instance_double(Dependant, becomes_16_on: submission_date + 1.year)] }
 
-        it "returns true" do
-          expect(calculated_result).to eq true
+          it "returns true" do
+            expect(calculated_result).to eq true
+          end
         end
-      end
 
-      context "with adult dependants, an employed applicant and no partner" do
-        let(:dependants) { [OpenStruct.new(becomes_16_on: submission_date - 1.year)] }
+        context "with child dependants, a student applicant and no partner" do
+          let(:dependants) { [instance_double(Dependant, becomes_16_on: submission_date + 1.year)] }
+          let(:applicant_gross_income_subtotals) do
+            instance_double(PersonGrossIncomeSubtotals, is_student?: true, employment_income_subtotals: instance_double(EmploymentIncomeSubtotals, in_work?: false))
+          end
 
-        it "returns true" do
-          expect(calculated_result).to eq false
+          it "returns true" do
+            expect(calculated_result).to eq true
+          end
         end
-      end
 
-      context "with child dependants, an employed applicant and an employed partner" do
-        let(:dependants) { [OpenStruct.new(becomes_16_on: submission_date + 1.year)] }
-        let(:partner) { OpenStruct.new(in_work?: true, is_student?: false) }
+        context "with adult dependants, an employed applicant and no partner" do
+          let(:dependants) { [OpenStruct.new(becomes_16_on: submission_date - 1.year)] }
 
-        it "returns true" do
-          expect(calculated_result).to eq true
+          it "returns true" do
+            expect(calculated_result).to eq false
+          end
         end
```

Comment:
This is strange - you've appear to have added added 'returns true' when it is clearly false. Finding these changes hard to follow - are they strictly necessary?

Reply:
Example description updated. I fixed the existing test and refactored a bit. Any suggestion what else we should add here?